src/api/odometry/odomMath.cpp

```diff
@@ -14,24 +14,47 @@ OdomMath::OdomMath() = default;
 
 OdomMath::~OdomMath() = default;
 
-QLength OdomMath::computeDistanceToPoint(const Point2d &ipoint, const OdomState &istate) {
-  const double xDiff = (ipoint.x - istate.x).convert(meter);
-  const double yDiff = (ipoint.y - istate.y).convert(meter);
-  return std::sqrt((xDiff * xDiff) + (yDiff * yDiff)) * meter;
+QLength OdomMath::computeDistanceToPoint(const Point2d &ipoint,
+                                         const OdomState &istate,
+                                         const StateMode &ipointMode) {
+  if (ipointMode == StateMode::CARTESIAN) {
+    const double xDiff = (ipoint.y - istate.x).convert(meter);
+    const double yDiff = (ipoint.x - istate.y).convert(meter);
+    return std::sqrt((xDiff * xDiff) + (yDiff * yDiff)) * meter;
+  } else {
+    const double xDiff = (ipoint.x - istate.x).convert(meter);
+    const double yDiff = (ipoint.y - istate.y).convert(meter);
+    return std::sqrt((xDiff * xDiff) + (yDiff * yDiff)) * meter;
+  }
 }
 
-QAngle OdomMath::computeAngleToPoint(const Point2d &ipoint, const OdomState &istate) {
-  const double xDiff = (ipoint.x - istate.x).convert(meter);
-  const double yDiff = (ipoint.y - istate.y).convert(meter);
-  return (std::atan2(yDiff, xDiff) * radian) - istate.theta;
+QAngle OdomMath::computeAngleToPoint(const Point2d &ipoint,
+                                     const OdomState &istate,
+                                     const StateMode &ipointMode) {
+  if (ipointMode == StateMode::CARTESIAN) {
+    const double xDiff = (ipoint.y - istate.x).convert(meter);
+    const double yDiff = (ipoint.x - istate.y).convert(meter);
+    return (std::atan2(yDiff, xDiff) * radian) - istate.theta;
+  } else {
+    const double xDiff = (ipoint.x - istate.x).convert(meter);
+    const double yDiff = (ipoint.y - istate.y).convert(meter);
+    return (std::atan2(yDiff, xDiff) * radian) - istate.theta;
+  }
 }
 
 std::pair<QLength, QAngle> OdomMath::computeDistanceAndAngleToPoint(const Point2d &ipoint,
```

Comment:
I think this function should delegate:
```cpp
return std::make_pair(computeDistanceToPoint(ipoint, istate, ipointMode), computeAngleToPoint(ipoint, istate, ipointMode));
```

Reply:
It doesn't delegate to save computing the diff's twice.